### PR TITLE
Remove redundant retry logic

### PR DESCRIPTION
**Pull request by altusgerona**

### CODE REVIEW

File: pyrmq/publisher.py [hunk 1]
```diff
@@ -130,9 +130,10 @@ def connect(self, retry_count=1) -> (BlockingConnection, BlockingChannel):
             return connection, channel
 
         except CONNECTION_ERRORS as error:
-            if not (retry_count % self.connection_attempts):
-                self.__send_reconnection_error_message(retry_count, error)
-                return
+            self.__send_reconnection_error_message(retry_count, error)
```
Comment (sibayanjasper) [inline pyrmq/publisher.py:133]: This will send message on every retry. We don't want that.

File: pyrmq/publisher.py [hunk 2]
```diff
@@ -130,9 +130,10 @@ def connect(self, retry_count=1) -> (BlockingConnection, BlockingChannel):
...
+            if not self.infinite_retry:
+                raise
```
Comment (sibayanjasper) [inline pyrmq/publisher.py:135]: raise error

File: pyrmq/publisher.py (hunk 1 above)
Comment (altusgerona) [reply pyrmq/publisher.py:133]: But `self.__create_connection` has already tried three times before this point.

Comment (sibayanjasper) [reply pyrmq/publisher.py:133]: Okay. Then retry_count here is not accurate. It will say it retried up to 1 times, 2 times, etc.

**Pull request merged**